core/templates/dev/head/pages/signup/Signup.js

```diff
@@ -173,9 +173,48 @@ oppia.controller('Signup', [
       $http.post(_SIGNUP_DATA_URL, requestParams).then(function() {
         window.location = window.decodeURIComponent(
           UrlService.getUrlParams().return_url);
-      }, function() {
+      }, function(rejection) {
+        var loggedOutInNewTabErrorMessage = (
+          'Sorry, you have been logged out [probably in another ' +
+          'window]. Please log in again. You will be redirected ' +
+          'to main page in a while!');
+        if (
+          rejection.data && (
+            rejection.data.error === loggedOutInNewTabErrorMessage)) {
```

Comment:
Use the status code instead to decide whether to show the modal.

Reply:
Done.

Reply:
If we just check the status code being 401, it will also be valid for the case when the csrf token has expired (401 is raised in that case too) and that will result in continue registration modal being shown in that case too. I feel it is fine to show the modal in both the cases. What do you think?